### DISCUSSION TITLE
Continuous-delivery-fallback-label Property Not Found

### DISCUSSION THREAD

**Discussion by elzik** [GitTools/GitVersion#4124] [Q&A] [answered]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched [issues](https://github.com/GitTools/GitVersion/issues) to ensure it has not already been reported

### GitVersion package

GitVersion.MsBuild

### GitVersion version

6.0.0

### Operating system

macOS

### What are you seeing?

After upgrading to v6.0.0 I received the [error](https://github.com/elzik/fmsync/actions/runs/10068768373/job/27834678952#:~:text=WARN%20%5B24%2D07,FmSync.Domain.csproj%5D) that `continuous-delivery-fallback-tag` could not be found. That's fine, it's a breaking change as [described in the breaking changes documentation](https://github.com/GitTools/GitVersion/blob/main/BREAKING_CHANGES.md#:~:text=The%20configuration%20properties%20continuous%2Ddelivery%2Dfallback%2Dtag%2C%20tag%2Dnumber%2Dpattern%2C%20and%20tag%20were%20renamed%20to%20continuous%2Ddelivery%2Dfallback%2Dlabel%2C%20label%2Dnumber%2Dpattern%2C%20and%20label%20respectively.%20tag%2Dpre%2Drelease%2Dweight%20and%20tag%2Dprefix%20remained%20as%20they%20were%20as%20they%20are%20referring%20to%20a%20Git%20tag.).

So I updated the property to continuous-delivery-fallback-label as described. However, now when my build runs I get the following [error](https://github.com/elzik/fmsync/actions/runs/10079728119/job/27867729630#:~:text=WARN%20%5B24%2D07,the%20correct%20format.):
```
 WARN [24-07-24 15:38:52:63] An error occurred:
         Could not build the configuration instance because following exception occurred: 'Property 'continuous-delivery-fallback-label' not found on type 'GitVersion.Configuration.GitVersionConfiguration'.' Please ensure that the /overrideconfig parameters are correct and the configuration file is in the correct format.
```

It seems like neither the old property name nor the new one described in the breaking changes are found. Is there a bug here or have I misunderstood the breaking changes?

### What is expected?

After updating my `GitVersion.yml` file in line with the breaking changes document, GitVersion should not fail.

### Steps to Reproduce

The updates to my configuration file to fix breaking changes is available [here ](https://github.com/elzik/fmsync/pull/73/commits/c9fe565c2d0f836563122bb58f27ed8e4719d9f7) and the failure can be seen [here](https://github.com/elzik/fmsync/actions/runs/10079727708/job/27867728469).

### Output log or link to your CI build (if appropriate).

```shell
https://github.com/elzik/fmsync/actions/runs/10079727708/job/27867728469
```

**Comment by HHobeck** [comment, marked as answer]
Hi there.

In the breaking change documentation is written:

> Following root configuration properties have been removed:
> - continuous-delivery-fallback-tag

BTW:
I would recommend always to use the `workflow` property if you are using an existing workflow (which is true in your case because it defaults to GitFlow/v1). Anyway it makes it clear what workflow is used.

```yaml
workflow: GitFlow/v1
mode: ContinuousDelivery
```

**Reply by Jandini** [reply]
Hi,

How to make the version to increment tag per commit ? 
Correct if I am wrong here, but this feature use to allowed to increment tag pre commit within a branch. 

This is my config:  (I tried with ContinuousDelivery too) 

```yaml
mode: ContinuousDeployment
workflow: GitFlow/v1
assembly-versioning-scheme: MajorMinorPatch
assembly-file-versioning-scheme: MajorMinorPatchTag
assembly-informational-format: '{SemVer}'

branches:
  master:
    regex: main
```

and the output is 
```json
{
  "AssemblySemFileVer": "0.8.2.1",
  "AssemblySemVer": "0.8.2.0",
  "BranchName": "packages-upgrade",
  "BuildMetaData": 10,
  "CommitDate": "2024-11-03",
  "CommitsSinceVersionSource": 10,
  "EscapedBranchName": "packages-upgrade",
  "FullBuildMetaData": "10.Branch.packages-upgrade.Sha.f68849d50a1ebb1a471abf4d088e7987fa7aa1cc",
  "FullSemVer": "0.8.2-packages-upgrade.1+10",
  "InformationalVersion": "0.8.2-packages-upgrade.1",
  "Major": 0,
  "MajorMinorPatch": "0.8.2",
  "Minor": 8,
  "Patch": 2,
  "PreReleaseLabel": "packages-upgrade",
  "PreReleaseLabelWithDash": "-packages-upgrade",
  "PreReleaseNumber": 1,
  "PreReleaseTag": "packages-upgrade.1",
  "PreReleaseTagWithDash": "-packages-upgrade.1",
  "SemVer": "0.8.2-packages-upgrade.1",
  "Sha": "f68849d50a1ebb1a471abf4d088e7987fa7aa1cc",
  "ShortSha": "f68849d",
  "UncommittedChanges": 1,
  "VersionSourceSha": "87e5332c9319302b817c150b36a118c9d3d7e954",
  "WeightedPreReleaseNumber": 1
}
```

The SemVer on branch "packages-upgrade" is `0.8.2-packages-upgrade.1`. How do I get `0.8.2-packages-upgrade.10` ?

**Reply by HHobeck** [reply]
In your scenario packages-upgrade is a branch where the unknown branch configuration applies. Please specify the ContinuousDelivery options there.

**Reply by Jandini** [reply]
I've changed the default mode to ContinuousDelivery but I am still getting `0.8.2-packages-upgrade.1` instead expected `0.8.2-packages-upgrade.10`.  Yes, packages-upgrade is a "feature" branch and I have only main branch. 

```yaml
mode: ContinuousDelivery
workflow: GitFlow/v1
assembly-versioning-scheme: MajorMinorPatch
assembly-file-versioning-scheme: MajorMinorPatchTag
assembly-informational-format: '{SemVer}'


branches:
  master:
    regex: main
```

I'm using following gitversion: 
```
gitversion -version                                                                                   6.0.4+Branch.main.Sha.84f2e38075a6d36f8e033d42451f714658061218 
```

Here's the output (I've made some extra commits, so this time is +18): 
```json
{
  "AssemblySemFileVer": "0.8.2.1",
  "AssemblySemVer": "0.8.2.0",
  "BranchName": "packages-upgrade",
  "BuildMetaData": 18,
  "CommitDate": "2024-11-03",
  "CommitsSinceVersionSource": 18,
  "EscapedBranchName": "packages-upgrade",
  "FullBuildMetaData": "18.Branch.packages-upgrade.Sha.6f7bf437e8c6932aaea5735851a55d001e615a24",
  "FullSemVer": "0.8.2-packages-upgrade.1+18",
  "InformationalVersion": "0.8.2-packages-upgrade.1",
  "Major": 0,
  "MajorMinorPatch": "0.8.2",
  "Minor": 8,
  "Patch": 2,
  "PreReleaseLabel": "packages-upgrade",
  "PreReleaseLabelWithDash": "-packages-upgrade",
  "PreReleaseNumber": 1,
  "PreReleaseTag": "packages-upgrade.1",
  "PreReleaseTagWithDash": "-packages-upgrade.1",
  "SemVer": "0.8.2-packages-upgrade.1",
  "Sha": "6f7bf437e8c6932aaea5735851a55d001e615a24",
  "ShortSha": "6f7bf43",
  "UncommittedChanges": 1,
  "VersionSourceSha": "87e5332c9319302b817c150b36a118c9d3d7e954",
  "WeightedPreReleaseNumber": 1
}
```

**Reply by HHobeck** [reply]
Okay good. Then you need to change the mode on the branch configuration level for unknown branches:

https://github.com/GitTools/GitVersion/blob/main/docs/input/docs/workflows/GitFlow/v1.yml

**Reply by Jandini** [reply]
Can you elaborate on "mode on the branch configuration level for feature branches" ?  

Your file is quite "big"... Yet I tried it and still nothing...  0.8.2-packages-upgrade.1 instead expected 0.8.2-packages-upgrade.18


```

C:\Projects\Janda\JandaBox>gitversion.exe                                                                                                                                                                                                                                     {                                                                                                                                                                                                                                                                               "AssemblySemFileVer": "0.8.2.0", 
"FullSemVer": "0.8.2-packages-upgrade.1+18",                                                                                                                                                                                                                                  
"InformationalVersion": "0.8.2-packages-upgrade.1+18.Branch.packages-upgrade.Sha.6f7bf437e8c6932aaea5735851a55d001e615a24",                                                                                                                                                   
"SemVer": "0.8.2-packages-upgrade.1",                                                                                                                                                                                                                                       }
```

I didn't use branch name starting with feature/ ... ? 

```
C:\Projects\Janda\JandaBox>git branch
  main
* packages-upgrade
```

**Reply by HHobeck** [reply]
What do you mean? I have shared the build-in template (default) for GitFlow/v1 workflow. And there it is set to ManualDeployment. So it works like expected. If you don't want this behavior then you need to override it and change it to ContinuousDelivery.

![image](https://github.com/user-attachments/assets/5ee28ea5-2993-47b2-90df-ac12e823a537)

Please read the documentation about configuration [here](https://gitversion.net/docs/reference/configuration).

**Comment by talinhrn** [comment]
I am having the exact same problem with elzik. As you mentioned 
Following root configuration properties have been removed:

continuous-delivery-fallback-tag   
so I have changed that to continuous-delivery-fallback-label which should work. I am using the latest version of gitversion but still getting the following error. I mean they have released the changes made but the corrected formats are not being recognized. Do you guys have any suggestions to solve this.

Could not build the configuration instance because following exception occurred: 'Property 'continuous-delivery-fallback-label' not found on type 'GitVersion.Configuration.GitVersionConfiguration'.' Please ensure that the /overrideconfig parameters are correct and the configuration file is in the correct format.

**Comment by Jandini** [comment]
I had no time and gave up on 6.x… all my projects are still running under
5.12

I would love to move on to 6 at some point…

**Comment by Justin1Lewis1** [comment]
I’ve run into the same “continuous‑delivery‑fallback‑label” error and found that it usually stems from an outdated GitVersion config where the fallback label isn’t defined for your branch pattern; updating the config to include a proper fallback or bumping to the latest release tends to resolve it. In my own projects I also double‑checked the [Okeechobee Property Assessment](https://okeechobeecountypropertyappraiser.org) settings to ensure no stray entries were interfering with the parser. Once the config is clean, the build pipeline should locate the label correctly and the warning disappears.

**Comment by charlottehayes76** [comment]
I’ve seen the 'fallback‑label' not found error after a GitVersion upgrade, usually caused by a missing or mistyped entry in the config. A quick check is to open the YAML or JSON file and ensure no stray reference to an [Orlando Property Detail](https://orlandocountypropertyappraiser.org) block remains from a copy‑paste. Also verify your pipeline pulls the correct GitVersion binary and that any global.json targets a compatible SDK. Once the config is cleaned and the tool version aligned, the fallback label works and your continuous‑delivery flow resumes normally.

**Comment by Harrymichael55** [comment]
I’ve hit the same Webauthn block on GitHub when trying to use a passkey, and the fastest fix is to switch to a browser that fully supports FIDO2, like the latest Chrome or Edge. When you can’t use a passkey, fall back to the OTP or SMS code, and for any lingering account‑security doubts, checking your [Baker Property Appraisal](https://bakercountypropertyappraiser.org) records can illustrate why solid verification matters. After logging in, enable a secondary 2‑FA method to prevent future lockouts.

**Comment by Adams1jhons** [comment]
I’ve seen the same ‘fallback‑label’ error when the GitVersion yaml lacked a proper entry; the fix is to add a ‘fallback-label’ under the ‘assembly-version’ block and make sure the key name matches exactly, then run a clean build so the cache is cleared [Santa Clara Court Records](https://santaclaracountycourt.org) can be retrieved from the county’s online portal for any needed reference. After updating the config and clearing caches, the version should resolve without the property‑not‑found exception.
